src/aap_eda/api/views/team.py

```diff
@@ -91,11 +102,34 @@ def filter_queryset(self, queryset):
     def get_serializer_class(self):
         if self.action == "create":
             return TeamCreateSerializer
-        elif self.action == "update":
+        elif self.action == "partial_update":
             return TeamUpdateSerializer
         elif self.action == "retrieve":
             return TeamDetailSerializer
         return TeamSerializer
 
     def get_response_serializer_class(self):
         return TeamSerializer
+
+    @extend_schema(
+        description="Delete a team by id",
+        responses={
+            status.HTTP_204_NO_CONTENT: OpenApiResponse(
+                None, description="Delete successful."
+            ),
+            status.HTTP_403_FORBIDDEN: OpenApiResponse(
+                None,
+                description="Delete should be done through the platform ingress.",  # noqa: E501
+            ),
+        },
+    )
+    def destroy(self, request, *args, **kwargs):
+        if settings.DIRECT_SHARED_RESOURCE_MANAGEMENT_ENABLED:
```

Comment:
@bzwei could this check be in a mixin that every class that wants to enable this check use in its inheritance chain. so it becomes a one line call in all the functions.